C7Engine/AI/ExplorerAI.cs

```diff
@@ -0,0 +1,131 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using C7GameData;
+using C7GameData.AIData;
+using C7Engine.Pathing;
+
+namespace C7Engine
+{
+	public class ExplorerAI
```

Comment:
It could be nice to have an interface for both AI and for AI data. Partially because it would be nice to treat AI polymorphically, but also because the current naming convention is slightly confusing for someone who isn't familiar with the code ie. we have `ExplorerAIData : UnitAI`, but I don't really expect implementations of `UnitAI` to contain data

Reply:
Yeah, I was thinking about that when I fell asleep last night and forgot about it when I woke up.  UnitAI is definitely confusing, I will rename that.  I should be able to get the AI to be polymorphic as well.

Reply:
I wound up spinning this (other than the UnitAI data rename) off into https://github.com/C7-Game/Prototype/pull/203, since it's a semi-decent chunk of changes and isn't really related to tile visibility/the initial explorer AI.  It is a nice improvement though, glad you mentioned it.